src/khoj/routers/api.py

```diff
@@ -276,6 +278,49 @@ async def transcribe(
     return Response(content=content, media_type="application/json", status_code=200)
 
 
+@api.get("/settings", response_class=Response)
+@requires(["authenticated"])
+def get_settings(request: Request, detailed: Optional[bool] = False) -> Response:
+    user = request.user.object
+    user_config = get_user_config(user, request, is_detailed=detailed)
+    del user_config["request"]
+
+    # Return config data as a JSON response
+    return Response(content=json.dumps(user_config), media_type="application/json", status_code=200)
+
+
+@api.patch("/user/name", status_code=200)
+@requires(["authenticated"])
+def set_user_name(
+    request: Request,
+    name: str,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    split_name = name.split(" ")
+
+    if len(split_name) > 2:
+        raise HTTPException(status_code=400, detail="Name must be in the format: Firstname Lastname")
```

Comment:
Is this rule necessary? I guess it's fine if people want to have their middle name's or something. We can just put a character limit.

Reply:
Not it's not actually required. We could just split into last word = last name and everthing else = first name and put a character limit. But maybe can deal with improving the username management separately?